DependencyInjection/Compiler/FormFactoryCompilerPass.php

```diff
@@ -45,7 +46,11 @@ public function process(ContainerBuilder $container)
         $factory = $container->getDefinition('sonata.core.form.extension.dependency');
         $factory->replaceArgument(1, $original->getArgument(1));
         $factory->replaceArgument(2, $original->getArgument(2));
-        $factory->replaceArgument(3, $original->getArgument(3));
+        try {
+            $factory->replaceArgument(3, $original->getArgument(3));
+        } catch (OutOfBoundsException $e) {
+            $factory->replaceArgument(3, []);
```

Comment:
Please use detection instead:

1. look for changes in the API of the framework bundle between 3.2 and 3.3
2. pick one
3. detect it (`method_exists` or whatever)
4. don't inject anything in case we are in 3.3